sandbox shell: too limited or should derivations provide own shell?

The minimal busybox shell may be too limiting-- notably, it's insufficient to build busybox itself since it doesn't support `$(( ))` expressions[1].

This can result in a rather unhappy system as of 700e21d6dac4683ef40f20127a6eb2a74b9bd8c6 .

While these features can be added (this one is controlled by `CONFIG_FEATURE_SH_MATH`),
I thought I'd bring up the issue to ensure things are working as expected and to address related questions:

* What functionality can derivations "expect" from /bin/sh?
* Is it a "bug" for derivations to overly rely on /bin/sh instead of using `stdenv.shell` (which once stdenv bootstrapping is complete should be a full bash shell?)

Depending on the answers, we may need a sprint to fix all broken packages over on nixpkgs before switching to 2.0.  Alternatively it may make sense to (sadface) revert back to using full bash until this is hammered out.

PS: https://github.com/dtzWill/nix/commit/1dcd3b80385c314178e26bf3646000d67c9f75e2 fixes the arith expression issue, FWIW.

<details><summary>[1] busybox build error (may need to use `--check` to force the build)</summary>
```
building
build flags: -j8 -l8 SHELL=/nix/store/i0ay05pqkbnvpfijm52mmlrp6kmkl80c-bash-4.4-p12/bin/bash CC=gcc\ -isystem\ /nix/store/ilr1hvanw8k63bxk4s1djmxbpx2zikii-musl-1.1.18/include\ -B/nix/store/ilr1hvanw8k63bxk4s1djmxbpx2zikii-musl-1.1.18/lib\ -L/nix/store/ilr1hvanw8k63bxk4s1djmxbpx2zikii-musl-1.1.18/lib
  SPLIT   include/autoconf.h -> include/config/*
  HOSTCC  applets/usage
  HOSTCC  applets/applet_tables
  GEN     include/bbconfigopts.h
  GEN     include/common_bufsiz.h
/build/busybox-1.28.0/scripts/generate_BUFSIZ.sh: line 56: syntax error: support for $((arith)) is disabled
make: *** [Makefile:857: include/config/MARKER] Error 2
make: *** Waiting for unfinished jobs....
  GEN     include/usage_compressed.h
  GEN     include/applet_tables.h include/NUM_APPLETS.h
  GEN     include/applet_tables.h include/NUM_APPLETS.h
  CC      applets/applets.o
  HOSTCC  applets/usage_pod
  LD      applets/built-in.o
builder for '/nix/store/k6qjvxpqr2idmnmvbvk3lmsrppzqvqp7-busybox-1.28.0.drv' failed with exit code 2
error: build of '/nix/store/k6qjvxpqr2idmnmvbvk3lmsrppzqvqp7-busybox-1.28.0.drv' failed
```
</details>

--- a/release-common.nix
+++ b/release-common.nix
@@ -6,10 +6,22 @@ rec {
     enableStatic = true;
     enableMinimal = true;
     extraConfig = ''
+      CONFIG_FEATURE_FANCY_ECHO y
+      CONFIG_FEATURE_SH_MATH y
+      CONFIG_FEATURE_SH_MATH_64 y
+
       CONFIG_ASH y
-      CONFIG_ASH_ECHO y
-      CONFIG_ASH_TEST y
       CONFIG_ASH_OPTIMIZE_FOR_SIZE y
+
+      CONFIG_ASH_ALIAS y
+      CONFIG_ASH_BASH_COMPAT y
+      CONFIG_ASH_CMDCMD y
+      CONFIG_ASH_ECHO y
+      CONFIG_ASH_GETOPTS y
+      CONFIG_ASH_INTERNAL_GLOB y
+      CONFIG_ASH_JOB_CONTROL y
+      CONFIG_ASH_PRINTF y
+      CONFIG_ASH_TEST y
     '';
   };
 

